src/main/java/it/systems/paymentreviser/entity/Case.java

```diff
@@ -0,0 +1,51 @@
+package it.systems.paymentreviser.entity;
+
+import it.systems.paymentreviser.dto.PaymentDTO;
+import it.systems.paymentreviser.enums.ResolutionStatus;
+import it.systems.paymentreviser.enums.ValidCurrency;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+public abstract class Case {
```

Comment:
Good idea to use polymorphism.

Reply:
Previous solution started to poke my eyes when revisoned it again.